connector/src/main/java/org/geysermc/connector/network/translators/collision/BoundingBox.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2019-2020 GeyserMC. http://geysermc.org
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @author GeyserMC
+ * @link https://github.com/GeyserMC/Geyser
+ */
+
+package org.geysermc.connector.network.translators.collision;
+
+import lombok.*;
+
+@Data
+@AllArgsConstructor
+public class BoundingBox {
+    private double middleX;
+    private double middleY;
+    private double middleZ;
+
+    private double sizeX;
+    private double sizeY;
+    private double sizeZ;
+
+    public void translate(double x, double y, double z) {
+        middleX += x;
+        middleY += y;
+        middleZ += z;
+    }
+
+    // Check intersection with no offset
+    // Currently never used, but will probably be useful in the future
+    /* public boolean checkIntersection(BoundingBox otherBox) {
```

Comment:
I think we should get rid of this - if we need it, we can check old git commits.

Reply:
Done